webapp/src/Controller/API/ContestController.php

```diff
@@ -332,59 +332,106 @@ public function setBannerAction(Request $request, string $cid, ValidatorInterfac
      *                 property="force",
      *                 description="Force overwriting the start_time even when in next 30s",
      *                 type="boolean",
+     *             ),
+     *             @OA\Property(
+     *                 property="scoreboard_thaw_time",
+     *                 description="The new unfreeze (thaw) time of the contest",
+     *                 type="string",
+     *                 format="date-time"
      *             )
      *         )
      *     )
      * )
      * @OA\Response(
      *     response="200",
-     *     description="Contest start time changed successfully",
+     *     description="Contest object if it changed",
      *     @OA\JsonContent(
-     *         type="string"
+     *         allOf={
+     *             @OA\Schema(ref=@Model(type=Contest::class)),
+     *             @OA\Schema(ref="#/components/schemas/Banner")
+     *         }
      *     )
      * )
+     * @OA\Response(
+     *     response="204",
+     *     description="The change was successful"
+     * )
      */
-    public function changeStartTimeAction(Request $request, string $cid): Response
+    public function changeTimesAction(Request $request, string $cid): Response
     {
         $contest  = $this->getContestWithId($request, $cid);
-        $now      = Utils::now();
+        $now      = (int)Utils::now();
         $changed  = false;
         if (!$request->request->has('id')) {
             $response = new JsonResponse('Missing "id" in request.', Response::HTTP_BAD_REQUEST);
```

Comment:
More a general comment, but can't we do an early return here and in similar places below, so that we don't have this long and deep nesting of if/elseif cases?

Reply:
Yes that’s better